module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1227,4 +1289,97 @@ public function __call($methodName, $params)
             'Cannot call method: ' . $this->getDriverClass() . '::' . $methodName
         );
     }
+
+    /**
+     * Get data for an ILS method from shared or session cache
+     *
+     * @param array $cacheSettings Cache settings
+     *
+     * @return ?array
+     */
+    protected function getCachedData(array $cacheSettings): ?array
+    {
+        $cacheKey = $cacheSettings['key'];
+        if ('shared' === $cacheSettings['storage']) {
+            return $this->getSharedCachedData($cacheKey);
+        }
+        if ($this->sessionCache && ($entry = $this->sessionCache[$cacheKey] ?? null)) {
+            if (time() - $entry['ts'] <= $cacheSettings['lifeTime']) {
+                return $entry['payload'];
+            }
+            unset($this->sessionCache[$cacheKey]);
+        }
+        return null;
+    }
+
+    /**
+     * Put data for an ILS method to shared or session cache.
+     *
+     * @param array $cacheSettings Cache settings
+     * @param array $data          Data to cache
+     *
+     * @return void
+     */
+    protected function putCachedData(array $cacheSettings, array $data): void
+    {
+        $cacheKey = $cacheSettings['key'];
+        if ('shared' === $cacheSettings['storage']) {
+            $this->putSharedCachedData($cacheKey, $data);
+            return;
+        }
+        if ($this->sessionCache) {
+            $this->sessionCache[$cacheKey] = [
+                'ts' => time(),
+                'payload' => $data,
+            ];
+        }
+    }
+
+    /**
+     * Get cache settings for a method
+     *
+     * @param string $methodName The name of the called method.
+     * @param array  $params     Array of passed parameters.
+     *
+     * @return ?array
+     */
+    protected function getCacheSettings($methodName, $params): ?array
```

Comment:
Worth adding types here?

```suggestion
    protected function getCacheSettings(string $methodName, array $params): ?array
```

Reply:
I dared not enforce types here because the existing methods don't have typing either. I think e.g. $params might work if passed null even if you're not supposed to. Or do you think this would be a good moment to add types to the existing methods too?

Reply:
Perhaps adding of types could be done in dev-11.0?

Reply:
Yes, I agree, if there's any risk of breaking something, the change should wait until 11.0.